internal/fs/fs.go

```diff
@@ -1783,6 +1790,12 @@ func (fs *fileSystem) renameDir(
 	}
 	pendingInodes = append(pendingInodes, oldDir)
 
+	// If old directory contains local (un-synced) files, rename operation is not supported.
+	entries := oldDir.LocalFileEntries(fs.localFileInodes)
```

Comment:
Do we need fs lock here? @vadlakondaswetha

Reply:
Ideally yes. what did we do for ReadDir ?